lib/compute/vm.js

```diff
@@ -415,6 +415,59 @@ VM.prototype.reset = function(callback) {
 };
 
 /**
+ * Set the metadata for this instance.
+ *
+ * @resource [Instances: setMetadata API Documentation]{@link https://cloud.google.com/compute/docs/reference/v1/instances/setMetadata}
+ *
+ * @param {object} metadata - New metadata.
+ * @param {function=} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {module:compute/operation} callback.operation - An operation object
+ *     that can be used to check the status of the request.
+ * @param {object} callback.apiResponse - The full API response.
+ *
+ * @example
+ * vm.setMetadata({
+ *   'startup-script': '...'
+ * }, function(err, operation, apiResponse) {
+ *   // `operation` is an Operation object that can be used to check the status
+ *   // of the request.
+ * });
+ */
+VM.prototype.setMetadata = function(metadata, callback) {
+  var self = this;
+
+  callback = callback || util.noop;
+
+  this.getMetadata(function(err, currentMetadata, apiResponse) {
```

Comment:
That's actually what this does (overwrites, instead of extends). The upstream API is designed that way, so I thought that might be the most expected use case. We have to get the metadata to use the "fingerprint", so the setMetadata operation only succeeds if the fingerprint represents the current state of metadata.